packages/router/src/grpc/view-protocol-server/address-by-index.test.ts

```diff
@@ -0,0 +1,65 @@
+import { beforeEach, describe, expect, test, vi } from 'vitest';
+import {
+  AddressByIndexRequest,
+  AddressByIndexResponse,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1alpha1/view_pb';
+import { ServicesInterface } from '@penumbra-zone/types';
+import { createContextValues, createHandlerContext, HandlerContext } from '@connectrpc/connect';
+import { ViewProtocolService } from '@buf/penumbra-zone_penumbra.connectrpc_es/penumbra/view/v1alpha1/view_connect';
+import { servicesCtx } from '../../ctx';
+import { addressByIndex } from './address-by-index';
+import { Address } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1alpha1/keys_pb';
+
+vi.mock('@penumbra-zone/wasm-bundler', () => vi.importActual('@penumbra-zone/wasm-nodejs'));
+
+describe('AddressByIndex request handler', () => {
+  let mockServices: ServicesInterface;
+  let mockCtx: HandlerContext;
+
+  beforeEach(() => {
+    mockServices = {
+      getWalletServices: () =>
+        Promise.resolve({
+          viewServer: {
+            fullViewingKey:
+              'penumbrafullviewingkey1vzfytwlvq067g2kz095vn7sgcft47hga40atrg5zu2crskm6tyyjysm28qg5nth2fqmdf5n0q530jreumjlsrcxjwtfv6zdmfpe5kqsa5lg09',
+          },
+        }),
+    } as ServicesInterface;
+
+    mockCtx = createHandlerContext({
+      service: ViewProtocolService,
+      method: ViewProtocolService.methods.addressByIndex,
+      protocolName: 'mock',
+      requestMethod: 'MOCK',
+      contextValues: createContextValues().set(servicesCtx, mockServices),
+    });
+  });
+
+  test('should successfully get AddressByIndex with default index', async () => {
+    const addressByIndexResponse = new AddressByIndexResponse(
+      await addressByIndex(new AddressByIndexRequest(), mockCtx),
+    );
```

Comment:
The response object seems to be wrapped by another response object? You can just do:

```typescript
    const response = await addressByIndex(new AddressByIndexRequest(), mockCtx);
    expect(response.address).toBeInstanceOf(Address);
```

Reply:
In this case, yes. In other cases wrap is necessary because `addressByIndex()` returns `AddressByIndexResponse | PartialMessage<AddressByIndexResponse>.`